lib/ui/painting/fragment_shader.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_FRAGMENT_SHADER_H_
+#define FLUTTER_LIB_UI_PAINTING_FRAGMENT_SHADER_H_
+
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/painting/image.h"
+#include "flutter/lib/ui/painting/image_shader.h"
+#include "flutter/lib/ui/painting/shader.h"
+#include "third_party/skia/include/effects/SkRuntimeEffect.h"
+#include "third_party/skia/include/core/SkShader.h"
+
+#include <string>
+
+namespace tonic {
+class DartLibraryNatives;
+}  // namespace tonic
+
+namespace flutter {
+
+class FragmentShader : public Shader {
+  DEFINE_WRAPPERTYPEINFO();
+  FML_FRIEND_MAKE_REF_COUNTED(FragmentShader);
+
+ public:
+  ~FragmentShader() override;
+  static fml::RefPtr<FragmentShader> Create();
+
+  void initWithSource(const std::string& source);
+
+  void setTime(float time);
+
+  void setImage(CanvasImage* image,
+                SkTileMode tmx,
+                SkTileMode tmy,
+                const tonic::Float64List& matrix4);
+
+  static void RegisterNatives(tonic::DartLibraryNatives* natives);
+
+ private:
+  float t_;
```

Comment:
Style nit; data members should appear below constructor/methods

https://google.github.io/styleguide/cppguide.html#Declaration_Order